db/migrate/20240507165222_create_llms.rb

```diff
@@ -0,0 +1,25 @@
+class CreateLlms < ActiveRecord::Migration[7.1]
+  def change
+    create_table :llms do |t|
+      t.string :name
+      t.text :description
+
+      t.timestamps
+    end
+
+    # Current models
+    up_only do
+      {
+        'open_ai_newest': 'Newest OpenAI Model',
+        'claude_newest': 'Newest Anthropic Model',
+        'claude-3-sonnet-20240229': 'Claude 3 Sonnet',
+        'gpt-4': 'ChatGPT 4',
+        'claude-3-opus-20240229': 'Claude 3 Opus',
+        'gpt-3.5-turbo-0125': 'ChatGPT 3.5 Turbo',
+        'gpt-4-turbo-2024-04-09': 'ChatGPT 4 Turbo'
+      }.each do |name, description|
+        LLM.create(name: name, description: description)
+      end
```

Comment:
* Let's include all the GPT-4 and GPT-3.5 models listed [on this page](https://platform.openai.com/docs/models/overview). It's a lot! But I've run into some surprising reasons why people preferred one very specific model, even an older one, so we might as well support them all.
* Let's name the models in this style where we put the numeric identifiers in parens, `GPT-4 Turbo`, `GPT-4 Turbo (2024-04-09)`, and `GPT-4 Preview (0125)`
* Let's include all the Anthropic models [from this table](https://docs.anthropic.com/claude/docs/models-overview#model-comparison) (all the 2.x and 1.x ones)
* Similarly name them `Claude 3 Opus (20240229)` and `Claude 2.1`

That's going to be a long list, and I really like the idea of having those special "newest" models. How about we name those:

* `'best_gpt': 'Best OpenAI GPT Model'` and `'best_claude': 'Best Claude Anthropic Model'` Partly I'm suggesting "best" because OpenAI sometimes releases a newer version of older, less powerful models, because of improved pricing. But I think this "magic model" of ours will always point to whatever the most capable model is, since that's what most people will want. Also, by beginning with "B" we ensure that you can just alphabetize this list in the drop-down and these two magic models are always listed first. :)

Reply:
Wow, ca. 30 entries. Is that going to work for people to scroll through a huge drop-down? 
So I update the get_next_message implementations in AIBackends::OpenAI/AIBackends::Anthropic to 
use the "best" model if assistant has such a model, with this task?

Reply:
Yea it’ll be a lot, but I think it’s okay. The magic ones will be first, then Claude ones, and so the long list will be last. I think the “power” of options trumps the long list and we can always improve UI in the future if needed.

Good point about making sure the “best” model gets used. I actually think that should live inside LLM. You can override “def model” so if it would return magic one then it conditionalizes to the hard coded “best” one. We’ll have to keep it updated as we add new ones, but that’s not too frequent.

Reply:
ChatGPT: Leaving out the "never widely rolled out" ones, e.g. gpt-4-32k-0613